depends/Makefile

```diff
@@ -94,12 +91,14 @@ binutils_path=$($($(host_arch)_$(host_os)_native_binutils)_prefixbin)
 toolchain_path=$($($(host_arch)_$(host_os)_native_toolchain)_prefixbin)
 final_build_id_long+=$(shell $(build_SHA256SUM) config.site.in)
 final_build_id+=$(shell echo -n $(final_build_id_long) | $(build_SHA256SUM) | cut -c-$(HASH_LENGTH))
-$(host_prefix)/.stamp_$(final_build_id): $(native_packages) $(packages) $(rust_crates)
+$(host_prefix)/.stamp_$(final_build_id): $(native_packages) $(packages)
 	$(AT)rm -rf $(@D)
 	$(AT)mkdir -p $(@D)
 	$(AT)echo copying packages: $^
 	$(AT)echo to: $(@D)
 	$(AT)cd $(@D); $(foreach package,$^, tar xf $($(package)_cached); )
+	$(AT)echo Vendoring crates...
+	$(AT)$(@D)/native/bin/cargo vendor --locked --manifest-path $(BASEDIR)/../Cargo.toml $(VENDORED_CRATES)
```

Comment:
Consider having an explicit option to prevent downloads here (#1663), which would use:
```
	$(AT)$(@D)/native/bin/cargo vendor --frozen --offline --manifest-path $(BASEDIR)/../Cargo.toml $(VENDORED_CRATES)
```
https://doc.rust-lang.org/cargo/commands/cargo-vendor.html

Reply:
It's somewhat tricky to implement this, because this particular pathway needs to simultaneously support offline builds (where the sources were pre-fetched via `make -C depends download`) and online builds (where that did not happen), and there is no usable way to set options for this. I'm instead relying on the fact that `cargo vendor` does not hit the network if nothing needs to be changed (and the Gitian build confirmed that this is working).

In any case, this particular bit of code will be altered to fix the other Gitian issue, so I'll see if there's maybe a better way to organise the logic within the depends system.

Reply:
Can the download target not write a zero-byte file, say `offline-build`, indicating that this use of `cargo vendor` should use `--frozen --offline`?

Reply:
I instead decided to rely on the fact that `make -C depends download` is only run when you want to do an offline build (like for Gitian). If the archive it generates exists, we use it and then run `cargo vendor --offline` to force that the archive is current. If the archive doesn't exist, we do an online vendor.